### PR TITLE
Feat  discussion

### PR DESCRIPTION
**Pull request by distalx**
This contains following functionalities.

Discussion:
Users can create a new discussion thread for a study group.
Users can also create a global discussion.
Members of the study group will receive a notification upon the creation of a discussion thread.

Voting:
User can upvote or downvote any discussion.

Subscribe/Unsubscribe:
Users can subscribe or unsubscribe to any discussion.
Users will by default be subscribed to a discussion by either creating one or responding to one.
TODO: subscribed users will receive notifications.

Recent Posts:
Visitor will see the 5 most recent posts on a Discussions page's right-hand sidebar section.

Active Posts
Users will be see the 5 most discussions they are participating in on the Discussion page's right-hand sidebar. If they are not participating in any discussions, they will see the global recent discussions.

Infinity Scroll
Infinite scroll for discussions.

Filter
A user or visitor will be able to filter discussions by newest, oldest, most commented, and least commented.

Responses
Users will be able to respond to any discussion.

Tags
Tags on the discussion page are clickable. On click, a user/visitor will only see discussions associated with that tag.

Slack
When they create a discussion topic, users can send an alert to channel by specifying a Slack channel.

Search
Users/visitors can search a discussion by topic or tag.

Report
Users can report/flag a discussion.

Moderator Actions
Site moderators can edit or remove reported or any discussions.

### CODE REVIEW

**Comment by nalbina** [comment]
I don't know how much sense it makes to pack so much features into one PR. I'd love to have it broken up more if thats feasible.

File: client/templates/discussion/all_discussions.js [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+
+  DocHead.setTitle(title);
+  DocHead.addMeta(metaInfo);
+
+  let instance = this;
+  instance.limit = new ReactiveVar(10);
+  instance.flag = new ReactiveVar(false);
+  instance.discussionFilter = new ReactiveVar('newest');
+
+  let projection = new Object();
```
Comment (nalbina) [inline client/templates/discussion/all_discussions.js]: why not `projection = {}` ?

Comment (distalx) [reply client/templates/discussion/all_discussions.js]: use of `new` operator provides more readability.

**Comment by distalx** [comment]
@nalbina, thank you for taking a look at it. this PR contains only one feature.

**Comment by distalx** [comment]
Merging this to the staging branch so can test out on staging.codebuddies.com, but the PR still needs a code review.

Also, planning on adding following things into a next iteration.

- Notifications
- Annotations
- Support for Emojis
- Responce Filter
- deep links for responses

**Pull request merged**